Controllers/TimeWorkedAndVacationController.cs

```diff
@@ -0,0 +1,154 @@
+{
+    public class TimeWorkedAndVacationController : Controller
+    {
+        private readonly stunning_robot_HRContext _context;
+
+        public TimeWorkedAndVacationController(stunning_robot_HRContext context)
+        {
+            _context = context;
+        }
+
+        // GET: TimeWorkedAndVacation
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.TimeWorkedAndVacation.ToListAsync());
+        }
+
+        // GET: TimeWorkedAndVacation/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var timeWorkedAndVacation = await _context.TimeWorkedAndVacation
+                .FirstOrDefaultAsync(m => m.TimeWorkedAndVacationId == id);
+            if (timeWorkedAndVacation == null)
+            {
+                return NotFound();
+            }
+
+            return View(timeWorkedAndVacation);
+        }
+
+        // GET: TimeWorkedAndVacation/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: TimeWorkedAndVacation/Create
+        // To protect from overposting attacks, enable the specific properties you want to bind to, for 
+        // more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("TimeWorkedAndVacationId,TotalNumberOfDaysWorked,TotalNumberOfAvailableVacationDays")] TimeWorkedAndVacation timeWorkedAndVacation)
+        {
+            if (ModelState.IsValid)
+            {
+                _context.Add(timeWorkedAndVacation);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            return View(timeWorkedAndVacation);
+        }
+
+        // GET: TimeWorkedAndVacation/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var timeWorkedAndVacation = await _context.TimeWorkedAndVacation.FindAsync(id);
+            if (timeWorkedAndVacation == null)
+            {
+                return NotFound();
+            }
+            return View(timeWorkedAndVacation);
+        }
+
+        // POST: TimeWorkedAndVacation/Edit/5
+        // To protect from overposting attacks, enable the specific properties you want to bind to, for 
+        // more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, [Bind("TimeWorkedAndVacationId,TotalNumberOfDaysWorked,TotalNumberOfAvailableVacationDays")] TimeWorkedAndVacation timeWorkedAndVacation)
+        {
+            if (id != timeWorkedAndVacation.TimeWorkedAndVacationId)
```

Comment:
You're getting close to the line of needing another class here. Traditionally, we create a second class called an ApplicationService. This class handles much of the non-web specific traffic such as data validation, calling repositories, ect. We'll discuss it wednesday.